src/app/clusters/time-format-localization-server/time-format-localization-logic.h

```diff
@@ -0,0 +1,74 @@
+/*
+ *    Copyright (c) 2025 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#pragma once
+
+#include <app/AttributeValueDecoder.h>
+#include <app/AttributeValueEncoder.h>
+#include <app/data-model-provider/ActionReturnStatus.h>
+#include <app/data-model-provider/MetadataTypes.h>
+#include <app/data-model-provider/OperationTypes.h>
+#include <app/persistence/AttributePersistenceProvider.h>
+#include <clusters/TimeFormatLocalization/Enums.h>
+#include <lib/core/CHIPError.h>
+#include <lib/support/ReadOnlyBuffer.h>
+#include <platform/DeviceInfoProvider.h>
+
+namespace chip {
+namespace app {
+namespace Clusters {
+
+class TimeFormatLocalizationLogic
+{
+public:
+    static constexpr TimeFormatLocalization::CalendarTypeEnum kDefaultCalendarType =
+        TimeFormatLocalization::CalendarTypeEnum::kBuddhist;
```

Comment:
spec does not seem to have a format and this seems like an odd default to have (maybe? no idea what the correct one if any is).

I expect previous applications to have used zap to select the right version. How about we keep the zap data and we query it in codegen integration to set it in the constructor?

It is unfortunate to need to pay the RAM cost for startup initialization, however this seems to be what applications would expect. In that case calnedar type and hour format would stay as RAM in zap.

Reply:
Added logic to read from RAM the values for HourFormat and CalendarType at startup.